### PR TITLE
Update the target tracker to use MedianFilter

### PR DESCRIPTION
**Pull request by mcm001**
Use the MedianFilter in TargetTracker. This should help filter random erroneous readings from solvePNP.

### CODE REVIEW

File: wpi/src/main/kotlin/org/ghrobotics/lib/vision/TargetTracker.kt [hunk 1]
```diff
@@ -145,16 +157,18 @@ open class TargetTracker(private val constants: TargetTrackerConstants) {
...
-                accumulatedY += sample.pose.translation.y
-                accumulatedAngle += sample.pose.rotation.radians
+            resetFilters()
+            var averageX = 0.0
+            var averageY = 0.0
+            var averageTheta = 0.0
+
+            samples.forEach {
+                averageX = xFilter.calculate(it.pose.translation.x)
+                averageY = xFilter.calculate(it.pose.translation.y)
```
Comment (prateekma) [inline wpi/src/main/kotlin/org/ghrobotics/lib/vision/TargetTracker.kt]: Should be `yFilter.calculate(...)`.

File: wpi/src/main/kotlin/org/ghrobotics/lib/vision/TargetTracker.kt [hunk 2]
```diff
@@ -145,16 +157,18 @@ open class TargetTracker(private val constants: TargetTrackerConstants) {
...
+                averageTheta = xFilter.calculate(it.pose.rotation.radians)
```
Comment (prateekma) [inline wpi/src/main/kotlin/org/ghrobotics/lib/vision/TargetTracker.kt]: Should be `thetaFilter.calculate(...)`.

File: wpi/src/main/kotlin/org/ghrobotics/lib/vision/TargetTracker.kt [hunk 3]
```diff
@@ -114,6 +115,11 @@ open class TargetTracker(private val constants: TargetTrackerConstants) {
         var averagePose = initialSample.pose
             private set
 
+        // a filter to average the pose of this target
+        private val xFilter = MedianFilter(constants.kMedianWindowSize);
+        private val yFilter = MedianFilter(constants.kMedianWindowSize);
+        private val thetaFilter = MedianFilter(constants.kMedianWindowSize);
```
Comment (prateekma) [inline wpi/src/main/kotlin/org/ghrobotics/lib/vision/TargetTracker.kt]: Semicolons redundant.

**Comment by mcm001** [comment]
Should be resolved in #2fefa7d

**Pull request merged**